src/objects/zcl_abapgit_object_odso.clas.abap

```diff
@@ -0,0 +1,349 @@
+        CREATE DATA lr_index_iobj  TYPE STANDARD TABLE OF ('BAPI6116II').
+      CATCH cx_sy_create_data_error.
+        zcx_abapgit_exception=>raise( |ODSO is not supported on this system| ).
+    ENDTRY.
+
+    ASSIGN lr_details->*     TO <ls_details>.
+    ASSIGN lr_infoobjects->* TO <lt_infoobjects>.
+    ASSIGN lr_navigation->*  TO <lt_navigation>.
+    ASSIGN lr_indexes->*     TO <lt_indexes>.
+    ASSIGN lr_index_iobj->*  TO <lt_index_iobj>.
+
+
+    io_xml->read( EXPORTING iv_name = 'ODSO'
+                  CHANGING  cg_data = <ls_details> ).
+
+    io_xml->read( EXPORTING iv_name = 'INFOOBJECTS'
+                  CHANGING  cg_data =  <lt_infoobjects> ).
+
+    io_xml->read( EXPORTING iv_name = 'NAVIGATION'
+                  CHANGING  cg_data =  <lt_navigation> ).
+
+    io_xml->read( EXPORTING iv_name = 'INDEXES'
+                  CHANGING  cg_data =  <lt_indexes> ).
+
+    io_xml->read( EXPORTING iv_name = 'INDEX_IOBJ'
+                  CHANGING  cg_data =  <lt_index_iobj> ).
+    TRY.
+        CALL FUNCTION 'BAPI_ODSO_CREATE'
+          EXPORTING
+            details              = <ls_details>
+          IMPORTING
+            odsobject            = lv_odso
+          TABLES
+            infoobjects          = <lt_infoobjects>
+            navigationattributes = <lt_navigation>
+            indexes              = <lt_indexes>
+            indexesinfoobjects   = <lt_index_iobj>
+            return               = lt_return.
+
+      CATCH  cx_sy_dyn_call_illegal_func.
+        zcx_abapgit_exception=>raise( |Necessary BW function modules not found or object not supported| ).
+    ENDTRY.
+
+    READ TABLE lt_return WITH KEY type = 'E' INTO ls_return.
+    IF sy-subrc = 0.
+      zcx_abapgit_exception=>raise( |Error when creating ODSO: { ls_return-message }| ).
+    ENDIF.
+
+    CALL FUNCTION 'BAPI_ODSO_ACTIVATE'
+      EXPORTING
+        odsobject = lv_odso
+      TABLES
+        return    = lt_return.
+
+    READ TABLE lt_return WITH KEY type = 'E' INTO ls_return.
+    IF sy-subrc = 0.
+      zcx_abapgit_exception=>raise( |Error when activating ODSO: { ls_return-message }| ).
+    ENDIF.
+
+    tadir_insert( iv_package ).
+
+    corr_insert( iv_package ).
+
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~exists.
+
+    DATA: lv_iobjnm TYPE c LENGTH 30.
+
+    SELECT SINGLE odsobject
+    FROM ('RSDODSO')
+    INTO lv_iobjnm
+    WHERE odsobject = ms_item-obj_name.
+
+    rv_bool = boolc( sy-subrc = 0 ).
+
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~get_comparator.
+    RETURN.
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~get_deserialize_steps.
+    APPEND zif_abapgit_object=>gc_step_id-abap TO rt_steps.
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~get_metadata.
+    rs_metadata = get_metadata( ).
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~is_active.
+
+    DATA: lv_dsona TYPE  c LENGTH 30.
+
+    lv_dsona = ms_item-obj_name.
+
+    rv_active = cl_rsd_odso=>factory( i_odsobject = lv_dsona )->is_active( ).
```

Comment:
looks like this class is a SAP_BW component, probably has to be called generically?